contracts/protocore/Protocore.sol

```diff
@@ -17,14 +17,13 @@ pragma solidity >=0.5.0 <0.6.0;
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 
 import "../consensus/CoconsensusModule.sol";
-import "../validator/ValidatorSet.sol";
 import "../version/MosaicVersion.sol";
 
 /**
  * @title Protocore abstract contract acting as a base contract for
  *        OriginProtocore and SelfProtocore contracts.
  */
-contract Protocore is MosaicVersion, ValidatorSet, CoconsensusModule {
+contract Protocore is MosaicVersion, CoconsensusModule {
```

Comment:
Why it's not ValidatorSet anymore?

Reply:
I think that the `ValidatorSet` should be in `SelfProtocore` and not `Protocore`.
`SelfProtocore` and `OriginProtocore` inherits the `Protocore` contract.
As per the specifications, `OriginProtocore` will not have validator sets and will rely on the `SelfProtocore` for any validator related queries. So the `SelfProtocore` will inherit `ValidatorSet` and not `Protocore`. We can discuss this point, else I can revert the changes.

Reply:
I do see your point. However, Casper game is played at Protocore (abstract). There we need to know does validator belongs to forward validator set of the given metablock height, or, what is the forward validators count at the given metablock height. All this are provided by ValidatorSet implementation. We can handle this probably in the https://github.com/mosaicdao/mosaic-1/issues/182 ticket by introducing a ValidatorSet interface.